configuration/NI.Analysis.targets

```diff
@@ -0,0 +1,95 @@
+		<IsNITestUtilitiesProject Condition="'$(IsNITestUtilitiesProject)' != 'True'">$(MSBuildProjectName.StartsWith("!TestUtilities."))</IsNITestUtilitiesProject>
+		<!-- The following is needed for misnamed TestUtilities that contain xaml because the generated project for the second compile appends random characters to the end of the project name -->
+		<IsNITestUtilitiesProject Condition="'$(IsNITestUtilitiesProject)' != 'True' and '$(IsNITestProject)' != 'True'">$(MSBuildProjectName.Contains(".TestUtilities"))</IsNITestUtilitiesProject>
+		<ProjectNameContainsTests>$(MSBuildProjectName.Contains("Tests"))</ProjectNameContainsTests>
+
+		<CodeAnalysisRuleSet>$(CodeAnalysisRuleSetDirectory)\NI.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.Tests.ruleset</CodeAnalysisRuleSet>
+		<CodeAnalysisRuleSet Condition="'$(IsNITestUtilitiesProjectCondition)' == 'True'">$(CodeAnalysisRuleSetDirectory)\NI.TestUtilities.ruleset</CodeAnalysisRuleSet>
+	</PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Enable Roslyn analyzers if they haven't been explicitly disabled, we're using Roslyn for CA/SA, the code needs to be validated -->
+    <EnableRoslynAnalyzers>True</EnableRoslynAnalyzers>
+    <IncludeStyleAnalyzers>True</IncludeStyleAnalyzers>
+    <IncludeCodeAnalyzers>True</IncludeCodeAnalyzers>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <StyleCopConfig>$(CodeAnalysisRuleSetDirectory)\stylecop.json</StyleCopConfig>
+    <NI1006_BannedMethods>$(CodeAnalysisRuleSetDirectory)\BannedMethods.xml</NI1006_BannedMethods>
+    <NI1004_LiteralExemptions>$(CodeAnalysisRuleSetDirectory)\LiteralExemptions.xml</NI1004_LiteralExemptions>
+
+    <!-- TODO: Determine whether we want to use the solution directory, the project directory, or both?-->
+    <NI1704_GlobalExceptions>$(SolutionDir)\CodeAnalysisDictionary.xml</NI1704_GlobalExceptions>
+
+    <!--AdditionalSpellingDictionary.dic contains the differences in words between the hunspell and Microsoft dictionaries-->
+    <NI1704_AdditionalSpellingDictionary>$(CodeAnalysisRuleSetDirectory)\NI1704_AdditionalSpellingDictionary.dic</NI1704_AdditionalSpellingDictionary>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(IncludeStyleAnalyzers)' == 'True'">
+    <!-- By adding the Link attribute, these files will not be shown in the Solution Explorer for .NET Core projects. -->
+    <AdditionalFiles Include="$(StyleCopConfig)" Condition="Exists('$(StyleCopConfig)')">
+      <Link>$(CodeAnalysisRuleSet)</Link>
+    </AdditionalFiles>
+
+    <!-- Microsoft's SA analyzers -->
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.dll"/>
+    <Analyzer Include="$(PkgStyleCop_Analyzers)\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll"/>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IncludeCodeAnalyzers)' == 'True'">
+    <RoslynAnalyzer Include="Microsoft.CodeQuality" />
```

Comment:
Explicitly adding Microsoft's analyzers is probably unnecessary (check Install.ps1)

Reply:
Removing this file for now, but we should revisit this for sure